src/coqLsp/coqLspTypes.ts

```diff
@@ -156,3 +156,13 @@ export class CoqLspError extends Error {
         this.name = "CoqLspError";
     }
 }
+
+export class CoqLspStartupError extends Error {
+    constructor(
+        message: string,
+        public readonly path: string
+    ) {
+        super(message);
+        this.name = "CoqLspStartupError";
```

Comment:
I'm not an expert in creating custom errors, so could you please share the idea behind this line? Also, I [found on the Internet](https://stackoverflow.com/a/41429145) that extending `Error` might actually break `instanceof` and related things: but there the suggested approach was to update the prototype
```TypeScript
Object.setPrototypeOf(this, CoqLspStartupError.prototype);
```

Reply:
Error names are used for example in our logging. E.g. `src/llm/llmServices/utils/generationsLogger/generationsLogger.ts`

Reply:
This is done to override parent's name.

Reply:
I made a series of experiments that showed no benefits of using `Object.setPrototypeOf(this, CoqLspStartupError.prototype);` and instanceof show no reasons to worry. Everything works fine, at least for reasonable typescript versions.